Submitted By: Alexander E. Patrakov
Date: 2006-07-13
Initial Package Version: 1.5.0
Origin: http://www.freebsd.org/cgi/cvsweb.cgi/~checkout~/ports/net-im/gaim/files/patch-src_protocols_oscar_oscar.c?rev=1.3&content-type=text/plain
Upstream Status: This is not a bug in GAIM.
Description: Do not escape HTML messages in ICQ if they begin with <HTML>.

This works around a bug with messages sent by the sim 0.9.3 client where they
would be displayed as: <HTML><BODY BGCOLOR="#FFFFFF">hello</BODY></HTML>

The bug is already fixed in sim 0.9.4, but there are lots of sim 0.9.3
installations in the wild.

--- gaim-1.5.0/src/protocols/oscar/oscar.c.orig	Wed Aug 25 21:29:00 2004
+++ gaim-1.5.0/src/protocols/oscar/oscar.c	Thu Aug 26 21:55:32 2004
@@ -3247,7 +3247,7 @@
 	 * Note: There *may* be some clients which send messages as HTML formatted -
 	 *       they need to be special-cased somehow.
 	 */
-	if (isdigit(gaim_account_get_username(account)[0]) && isdigit(userinfo->sn[0])) {
+	if (isdigit(gaim_account_get_username(account)[0]) && isdigit(userinfo->sn[0]) && g_strncasecmp(tmp, "<HTML>", 6)) {
 		/* being recevied by ICQ from ICQ - escape HTML so it is displayed as sent */
 		gchar *tmp2 = gaim_escape_html(tmp);
 		g_free(tmp);
